### ISSUE TITLE
Add an option to show the time left

### ISSUE THREAD

**Issue by tomassasovsky** [suragch/audio_video_progress_bar#7]
I want to use this package to display the progress on a video playing, and it would be really useful to have an option to, instead of displaying the full length of the video, display the time left.

**Comment by suragch** [comment]
What would the API look like for that? It's not that it would be hard to implement, but I'm just worried that if I keep adding options the API will become confusing to use.

In the meantime you can turn off the labels like this:

```dart
ProgressBar(
  ...
  timeLabelLocation: TimeLabelLocation.none,
);
```
And then implement it yourself with Text widgets.

**Comment by tomassasovsky** [comment]
Maybe add a boolean property called `showTimeLeft` or something like that and if set to true you just show the time left?

**Comment by tomassasovsky** [comment]
Here is a function to transform the duration to a String nicely. You can call it like this:

timeLeft = durationLeft(progress - duration);


```
String durationLeft(Duration duration) {
  String twoDigits(int n) => n.toString().padLeft(2, '0');

  duration = duration.abs();

  String twoDigitSeconds = twoDigits(duration.inSeconds.remainder(60));
  String twoDigitMinutes = twoDigits(duration.inMinutes.remainder(60));
  String twoDigitHours = twoDigits(duration.inHours);

  if (duration.inHours > 0)
    return '$twoDigitHours:$twoDigitMinutes:$twoDigitSeconds';
  else
    return '$twoDigitMinutes:$twoDigitSeconds';
}
```
I hope this helps.

**Comment by suragch** [comment]
Would the time left be shown as a negative number or as a positive number? It seems like I've usually seen it as negative:

https://cdn.dribbble.com/users/1963/screenshots/296133/screen_shot_2011-10-18_at_5.41.24_pm__400_x_300_.png

**Comment by tomassasovsky** [comment]
I've seen it as both negative and positive. Maybe negative?

**Comment by superakabo** [comment]
> I've seen it as both negative and positive. Maybe negative?

It should be negative

**Comment by suragch** [comment]
I'm thinking not to add this as a boolean. You two may want the time left as a negative but someone else may want it as a positive. Then we would have to add another boolean for that. 

I'm thinking something like this:

```dart
timeLabelType: TimeLabelType.remainingTime,
```

**Comment by suragch** [comment]
I've pushed a change to the master branch where you can set the time label type for time remaining:

```dart
ProgressBar(
  ...
  timeLabelType: TimeLabelType.timeRemaining,
);
```

How does this look?

<img width="302" alt="Screen Shot 2021-04-30 at 6 47 53 PM" src="https://user-images.githubusercontent.com/10300220/116685385-d3717d80-a9e4-11eb-8e0f-4b9cda9e95d9.png">

<img width="292" alt="Screen Shot 2021-04-30 at 6 48 23 PM" src="https://user-images.githubusercontent.com/10300220/116685421-e5ebb700-a9e4-11eb-92ba-bb0687f78bee.png">

**Comment by superakabo** [comment]
> I've pushed a change to the master branch where you can set the time label type for time remaining:
> 
> ```dart
> ProgressBar(
>   ...
>   timeLabelType: TimeLabelType.timeRemaining,
> );
> ```
> 
> How does this look?
> 
> <img alt="Screen Shot 2021-04-30 at 6 47 53 PM" width="302" src="https://user-images.githubusercontent.com/10300220/116685385-d3717d80-a9e4-11eb-8e0f-4b9cda9e95d9.png">
> 
> <img alt="Screen Shot 2021-04-30 at 6 48 23 PM" width="292" src="https://user-images.githubusercontent.com/10300220/116685421-e5ebb700-a9e4-11eb-92ba-bb0687f78bee.png">

Great!

**Comment by suragch** [comment]
Published now as version 0.4.0
